db/users.go

```diff
@@ -77,7 +79,7 @@ func (dbc *DatabaseContext) UpdatePrincipal(newInfo PrincipalConfig, isUser bool
 	var user auth.User
 	authenticator := dbc.Authenticator()
 	if isUser {
-		isValid, reason := newInfo.IsPasswordValid(dbc.AllowEmptyPassword)
+		isValid, reason := newInfo.IsPasswordValid(dbc.AllowEmptyPassword || allowReplace)
```

Comment:
What is the allowReplace change accomplishing here?  It looks like we are doing the following:
 - on an API call, allowReplace="request != POST"
 - for config users, allowReplace = isGuest
I'm not clear on how that interacts with 'allowEmptyPass'